azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -100,6 +103,7 @@ public JdbcProjectImpl(final Props props, final DatabaseOperator databaseOperato
     }
     int maxFlowFileSizeInMB = props.getInt(MAX_FLOW_FILE_SIZE_KEY, MAX_FLOW_FILE_SIZE_IN_MB_DEFAULT);
     maxFlowFileSizeInBytes = maxFlowFileSizeInMB * 1024 * 1024;
+    uploadProjectLockFeatureEnabled = props.get(AZKABAN_UPLOAD_PRIVILEGE_USER) != null;
```

Comment:
How about changing
"props.get(AZKABAN_UPLOAD_PRIVILEGE_USER) != null;"
to 
props.get(AZKABAN_UPLOAD_PRIVILEGE_USER, false);

Reply:
Ah, I see where the confusion is, AZKABAN_UPLOAD_PRIVILEGE_USER is not a boolean config as it might look like it is. It has the real user define in our config, and I am using it as a button to check if the feature is turned on or off.
When we want to roll out this feature, we can pass the value to job props otherwise we don't.